contracts/SnappAuction.sol

```diff
@@ -56,9 +75,66 @@ contract SnappAuction is SnappBase {
         return auctions[slot].appliedAccountStateIndex != 0;
     }
 
+    function getStandingOrderHash(uint16 userId, uint128 nonce) public view returns (bytes32) {
+        return reservedAccountOrders[userId][nonce].orderHash;
+    }
+    
+    function getStandingOrdervalidFrom(uint16 userId, uint128 nonce) public view returns (uint) {
+        return reservedAccountOrders[userId][nonce].validFrom;
+    }
+
+    function getStandingOrderValidityTo(uint16 userId, uint128 nonce) public view returns (uint) {
+        return reservedAccountOrders[userId][nonce].validTo;
+    }
+
     /**
      * Auction Functionality
      */
+    function placeStandingSellOrder(
+        uint8[] memory buyTokens,
+        uint8[] memory sellTokens,
+        uint128[] memory buyAmounts,
+        uint128[] memory sellAmounts
+    ) public onlyRegistered() {
+        
+        // Update Auction Hash based on request
+        uint16 accountId = publicKeyToAccountMap(msg.sender);
+        require(accountId <= AUCTION_RESERVED_ACCOUNTS, "Accout is not a reserved account");
+
+        bytes32 orderHash;
+        uint numOrders = buyToken.length;
+        require(numOrders <= AUCTION_RESERVED_ACCOUNT_BATCH_SIZE, "Too many orders");
+        
+        if (
+            auctionIndex == MAX_UINT ||
+            block.timestamp > (auctions[auctionIndex].creationTimestamp + 3 minutes)
+        ) {
+            createNewAuctionBatch();
+        }
+
+        for (uint i = 0; i < numOrders; i++) {
+            // Must have 0 < tokenId < MAX_TOKENS anyway, so may as well ensure registered.
+            require(buyTokens[i] < numTokens, "Buy token is not registered");
+            require(sellTokens[i] < numTokens, "Sell token is not registered");
+            orderHash = sha256(
+                abi.encodePacked(
+                    orderHash,
+                    encodeOrder(accountId, buyTokens[i], sellTokens[i], buyAmounts[i], sellAmounts[i])
+                )
+            );
+            emit StandingSellOrder(auctionIndex, accountId, buyTokens[i], sellTokens[i], buyAmounts[i], sellAmounts[i]);
+        }
+        uint128 currentNonce = standingOrderNonce[accountId];
+        if (auctionIndex > 0) {
```

Comment:
Would it be better/simpler to override the currentNonce if the `validFrom == auctionIndex` (that is multiple updates per block)?